tests/main/refresh-mode-ignore-running/task.yaml

```diff
@@ -17,14 +17,17 @@ execute: |
 
   trap 'touch stamp' EXIT
 
+  # refresh app awareness prevents the snap from being refreshed while it's running
   test-snapd-refresh.sh -c "while [ ! -e stamp ]; do sleep 1; done" &
-  not snap install --dangerous ./test-snapd-refresh_1_all.snap
+  snap install --dangerous ./test-snapd-refresh_1_all.snap 2>&1 | tr -s "\n" " " | MATCH ".*snap \"test-snapd-refresh\" has running apps \(sh\), pids:.*"
```

Comment:
I would do this in two lines:
```
not snap install --dangerous ./test-snapd-refresh_1_all.snap 2>error.txt
tr -s "\n" " " error.txt | MATCH ".*snap \"test-snapd-refresh\" has running apps \(sh\), pids:.*"
```
(note that I did not actually test that I did not do typos here)

mainly to ensure that the "snap install" also errors with an non-zero exit code (we test this elsewhere too for a different error case so one might argue it's redundant). And then the second line tests the actually error message.

The shell will by default ignore any failures on the left hand side of the pipe if we would not error in snap instal this new test would not catch that.